src/mca/bfrops/base/base.h

```diff
@@ -158,47 +160,306 @@ PMIX_EXPORT extern pmix_bfrops_globals_t pmix_bfrops_globals;
+        case PMIX_INT16:                                            \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_SIGNED(int16_t, s, d);    \
+            break;                                                  \
+        case PMIX_UINT16:                                           \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_UNSIGNED(uint16_t, s, d); \
+            break;                                                  \
+        case PMIX_INT:                                              \
+        case PMIX_INT32:                                            \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_SIGNED(int32_t, s, d);    \
+            break;                                                  \
+        case PMIX_UINT:                                             \
+        case PMIX_UINT32:                                           \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_UNSIGNED(uint32_t, s, d); \
+            break;                                                  \
+        case PMIX_INT64:                                            \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_SIGNED(int64_t, s, d);    \
+            break;                                                  \
+        case PMIX_SIZE:                                             \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_UNSIGNED(size_t, s, d);   \
+            break;                                                  \
+        case PMIX_UINT64:                                           \
+            PMIX_BFROPS_PACK_FLEX_CONVERT_UNSIGNED(uint64_t, s, d); \
+            break;                                                  \
+        default:                                                    \
+            (r) = PMIX_ERR_BAD_PARAM;                               \
+    }                                                               \
+} while(0)
+
+/**
+ * Unpacking conversion from a flexible representation to a
+ * signed integer value.
+ *
+ * type - C-type of a signed integer value
+ *
+ * val - flexible representation (uint64_t)
+ *
+ * ptr - pointer to a 64-bit output buffer for the upacked value
+ *
+ * (see a comment to `pmix_bfrops_pack_flex` for additional details)
+ */
+#define PMIX_BFROPS_UNPACK_FLEX_CONVERT_SIGNED(type, val, ptr)  \
+do {                                                            \
+    type __tbuf = 0;                                            \
+    uint64_t __tmp = val;                                       \
+    int sign = (__tmp) & 1;                                     \
+    __tmp >>= 1;                                                \
+    if (sign) {                                                 \
+        __tmp = ~__tmp;                                         \
+    }                                                           \
+    __tbuf = (type)__tmp;                                       \
+    memcpy(ptr, &__tbuf, sizeof(type));                         \
+} while (0)
```

Comment:
(very minor suggestion) It would be nice if the `PMIX_BFROPS_PACK_FLEX_CONVERT_SIGNED ` and `PMIX_BFROPS_UNPACK_FLEX_CONVERT_SIGNED` were next to each other in the code (same with `UNSIGNED`). That way it's easier to see them together since they are paired.

Reply:
Sure,
@karasevb please do

Reply:
Done with commit 5b423caaddabcb24337fc836b77e4704a7d438f4